src/protocol.cpp

```diff
@@ -87,17 +87,33 @@ CInv::CInv()
 {
     type = 0;
     hash.SetNull();
+    hashAux.SetNull();
 }
 
 CInv::CInv(int typeIn, const uint256& hashIn)
 {
+    assert(typeIn != MSG_WTX);
     type = typeIn;
     hash = hashIn;
+    if (typeIn == MSG_TX) {
+        hashAux = LEGACY_TX_AUTH_DIGEST;
+    } else {
+        hashAux.SetNull();
+    }
+}
+
+CInv::CInv(int typeIn, const uint256& hashIn, const uint256& hashAuxIn)
+{
+    type = typeIn;
```

Comment:
Does `hashAuxIn` make sense for any message types other than `MSG_WTX`? Should this assert on other usage? And, if so, does it make sense to make these constructors private and use static factory methods instead?